surface/include/pcl/surface/ear_clipping.h

```diff
@@ -83,11 +83,12 @@ namespace pcl
       void
       triangulate (const Vertices& vertices, PolygonMesh& output);
 
-      /** \brief Compute the signed area of a polygon. 
-        * \param[in] vertices the vertices representing the polygon 
+      /** \brief Triangulate one polygon, assume the vertices are clockwise. 
+        * \param[in] vertices the set of vertices
+        * \param[out] output the resultant polygonal mesh
         */
-      float
-      area (const std::vector<uint32_t>& vertices);
+      size_t
+      triangulate (std::vector<uint32_t>& vertices, PolygonMesh& output);
```

Comment:
@taketwo what's our policy on API changes of non-virtual protected methods? Purist side: still breaks the API since our consumers might have derived classes from this and use this method. Pragmatic side: unlikely.

Reply:
I think generallry we can approach "protected" API stability in a relaxed way, i.e. accept breakage without deprecation cycle. That said, isn't it super easy to keep `area()` method around and thus definitely not break anyone's code?

Reply:
I haven't checked yet. I decided to read a little on ear clipping triangulation before having a look with some proper eyes to the PR. I was failing to get the jargon of the OP. ^^

Reply:
The code is not used at all, so there's no reason to keep it around. I would remove it.